api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +565,148 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
```

Comment:
I think this can be simpler:

```java
    private GeometryType(String crs) {
      Preconditions.checkArgument(crs == null || !crs.isEmpty(), "Invalid CRS: (empty string)");
      this.crs = CRS84.equalsIgnoreCase(crs) ? null : crs;
```

Reply:
Use this simplified version.